tests/cli/test_suite.py

```diff
@@ -3081,3 +3087,719 @@ def test_suite_new_profile_runs_notebook_opens_jupyter(
         my_caplog=caplog,
         click_result=result,
     )
+
```

Comment:
❤️ This is superb -- and educational, too!  Thank you, @anthonyburdi for adding/refactoring/optimizing the code!
P.S.: Should not be too much work if the argument processors are moved to a `toolkit` type of a module (perhaps the `toolkit.py` itself).  Also, if PyCharm flags the importing of private methods, let us add the appropriate PyChard suppress macro.  Thanks!